web/src/typing.ts

```diff
@@ -82,6 +82,10 @@ function send_stream_typing_notification(
     topic: string,
     operation: "start" | "stop",
 ): void {
+    const stream = stream_data.get_sub_by_id(stream_id)!;
+    if (!stream_data.can_post_messages_in_stream(stream)) {
+        return;
+    }
```

Comment:
Mentioned in commit message too:

We did not add any checks for `message_edit_typing` since it is not
possible to get to that state in the UI for an archived channel's
messages.